server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -0,0 +1,157 @@
+
+    // Calling the API
+    String response = new SentimentRequest(text).getResponse();
+
+    // Extracting the sentiment
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonObject sentimentObject = jObject.getAsJsonObject("documentSentiment");
+
+    MagnitudeAndScore nlpResults =
+        new MagnitudeAndScore(
+            Double.valueOf(sentimentObject.get("magnitude").toString()),
+            Double.valueOf(sentimentObject.get("score").toString()));
+
+    return nlpResults;
+  }
+
+  private HashMap<String, String> analyzeWithPerspective(String text) throws IOException {
+
+    ArrayList<String> attributes =
+        new ArrayList<>(
+            Arrays.asList(
+                "TOXICITY",
+                "IDENTITY_ATTACK",
+                "INSULT",
+                "PROFANITY",
+                "THREAT",
+                "SEXUALLY_EXPLICIT",
+                "FLIRTATION"));
+
+    // Calling the API
+    PerspectiveRequest perspectiveRequest = new PerspectiveRequest(text, attributes);
+    String response = perspectiveRequest.getResponse();
+
+    // Extracting the results
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonObject attributeObject = jObject.getAsJsonObject("attributeScores");
+
+    // Placing them in a Hash Map
+    HashMap<String, String> perspectiveResults = new HashMap<String, String>();
+    for (String el : attributes) {
+      perspectiveResults.put(
+          el,
+          attributeObject
+              .getAsJsonObject(el)
+              .getAsJsonObject("summaryScore")
+              .get("value")
+              .toString());
+    }
+
+    return perspectiveResults;
+  }
+
+  private String getLyrics(String trackName, String artistName) throws IOException {
+
+    /** Getting the track id from Musixmatch */
+
+    // Calling the API
+    MusixParamBuilder params = new MusixParamBuilder("track.search", trackName, artistName);
+    MusixRequest musixReq = new MusixRequest("track.search", params.filterParamString());
+    String response = musixReq.grabResponse();
+
+    // Extracting the track id
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonArray resultList =
+        jObject.getAsJsonObject("message").getAsJsonObject("body").getAsJsonArray("track_list");
+    Iterator<JsonElement> iterator = resultList.iterator();
+    JsonElement firstResult = iterator.next();
+    String trackId =
+        firstResult.getAsJsonObject().getAsJsonObject("track").get("track_id").toString();
+
+    /** Getting the lyrics from the extracted id */
+
+    // Calling the API
+    params = new MusixParamBuilder("track.lyrics.get", trackId);
+    musixReq = new MusixRequest("track.lyrics.get", params.filterParamString());
+    response = musixReq.grabResponse();
```

Comment:
Note that these lines that "call the API" are almost identical to lines 120-122.  Maybe add a helper function that calls the API?

So, this way in your helper functions that (1) gets track id from musixmatch and (2) get lyrics from extracted id, you just have to call this helper function that "calls the API" in both (1) and (2).